### PR TITLE
Frontmatter style guide fixes

### PR DESCRIPTION
**Pull request by austin-schaefer**
* Clarifies what all the docs site frontmatter fields do
* Corrects a bunch of inaccuracy about old content types
* Adds definitions for new content types that weren't accurate
* Sweeps docs to remove a bunch of unused frontmatter fields to avoid future confusion

**Reviewer**: Please focus on the "Frontmatter and page templates" doc, the rest of this is just sweep work. I want to ensure the changed doc is helpful and accurate for writers

### CODE REVIEW

File: src/content/docs/style-guide/writing-docs/processes-procedures/frontmatter-page-templates.mdx [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+  - /docs/new-relic-only/tech-writer-style-guide/processes-procedures/using-content-editors
+  - /docs/style-guide/processes-procedures/use-content-types-text-formats
+  - /docs/style-guide/writing-docs/processes-procedures/use-content-types-text-formats/
+---
+
+This style guide page describes the standard frontmatter for docs, and also outlines the various types of content we maintain on the site. 
+
+## Docs frontmatter and metadata [#meta-fields]
+
+The top of every doc begins with a set of metadata. Read on For more about this metadata content:
```
Comment (akristen) [inline src/content/docs/style-guide/writing-docs/processes-procedures/frontmatter-page-templates.mdx:19]:
```suggestion
The top of every doc begins with a set of metadata. Read on for more about this metadata content:
```

File: src/nav/style-guide.yml [hunk 2]
```diff
@@ -100,7 +100,7 @@ pages:
               - title: Edit the docs site structure and nav
                 path: /docs/style-guide/writing-docs/processes-procedures/understand-edit-docs-site-structure
               - title: Frontmatter and page templates
-                path: /docs/style-guide/writing-docs/processes-procedures/use-content-types-text-formats
+                path: /docs/style-guide/writing-docs/processes-procedures/frontmatter-page-templatess
```
Comment (akristen) [inline src/nav/style-guide.yml:103]: hehe

File: src/content/docs/style-guide/writing-docs/processes-procedures/frontmatter-page-templates.mdx [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+tags:
+  - Tech writer style guide
+  - Processes and procedures
+redirects:
+  - /docs/new-relic-only/style-guide/processes-and-procedures/using-content-editors
+  - /docs/new-relic-only/style-guide/processes-procedures/using-content-editors
+  - /docs/new-relic-only/advanced-style-guide/processes-procedures/using-content-editors
+  - /docs/new-relic-only/tech-writer-style-guide/processes-procedures/using-content-editors
+  - /docs/style-guide/processes-procedures/use-content-types-text-formats
+  - /docs/style-guide/writing-docs/processes-procedures/use-content-types-text-formats/
```
Comment (akristen) [inline src/content/docs/style-guide/writing-docs/processes-procedures/frontmatter-page-templates.mdx:12]: possibly this trailing `/`

**Review by akristen — DISMISSED** [review]
This looks good to me! I reviewed the frontmatter doc and it built fine, tables rendered correctly, and the copy is good. Some nitpicky things, but overall i think it's good to merge as is :)

**Review by nbaenam — APPROVED** [review]
I fixed a typo. Other than that it's perfect!

**Pull request merged**